Source/WarpX.cpp

```diff
@@ -247,6 +247,25 @@ WarpX::WarpX ()
     // at different levels (the stencil depends on c*dt/dz)
     nci_godfrey_filter_exeybz.resize(nlevs_max);
     nci_godfrey_filter_bxbyez.resize(nlevs_max);
+
+    // Sanity checks. Must be done after calling the MultiParticleContainer
+    // constructor, as it reads additional parameters
+    // (e.g., use_fdtd_nci_corr)
+
+#ifndef WARPX_USE_PSATD
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        not ( do_pml && do_nodal ),
+        "PML + do_nodal for finite-difference not implemented"
+        );
+#endif
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
+        not ( do_dive_cleaning && do_nodal ),
+        "divE cleaning + do_nodal not implemented"
+        );
+#ifdef WARPX_USE_PSATD
+    BL_ASSERT(use_fdtd_nci_corr == 0);
```

Comment:
Does it make sense to mix `BL_ASSERT` and `AMREX_ALWAYS_ASSERT` in code?

I would try to get rid of all the `BL_` stuff where possible, since it's horribly confusing to new developers :)

Reply:
`BL_ASSERT` appears in many places in the code, is I think there should be a separate PR just to replace all `BL_ASSERT` with something else. But I can change these also if you prefer.

Reply:
I did the modification :)